0015-automatic-cluster-upgrades/0015-automatic-cluster-upgrades.md

```diff
@@ -0,0 +1,95 @@
+# RFC 0015 - Automatic Cluster Upgrades
+
+## User stories
+
+- As a user, and also as KaaS team provider, I would like to have the clusters running in the latest version automatically, and not worry about CVEs or EOL of the cluster components.
+
+- As a user, I would like to define maintenance windows to ensure clusters are not upgraded outside of these hours.
+
+- As a user I would like to mark a cluster as frozen and block upgrades for certain period of time.
+
+- As a user I want to define a policy that enforces some environments are upgraded before than others to an agreed version.
+
+- As a user I want to schedule a upgrade for a certain day/time.
+
+- As a user, and also as KaaS team provider, I want to capture the result of upgrade.
+
+## Reasoning
+
+Our customers need to plan every upgrade on their side and manage the upgrades for every GS version. Every customer has their own requirement when they need to schedule upgrades (freeze periods, maintenance windows,...). At the same time, Cloud Native projects and Giant Swarm are releasing more versions every time. It provokes a lot of time preparing and executing these upgrades which is not usually given much value to any of the stakeholders. 
```

Comment:
```suggestion
Our customers need to plan every upgrade on their side and manage the upgrades for every GS version. Every customer has their own requirement when they need to schedule upgrades (freeze periods, maintenance windows,...). At the same time, Cloud Native projects and Giant Swarm are continuously releasing new versions. This means a lot of time is consumed preparing and executing these upgrades which is not usually given much value by any of the stakeholders. 
```

Hope I got this right, you meant that the stakeholders are not giving much value to upgrades, right?

Reply:
you got it right